backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/OptimizationPipeline.kt

```diff
@@ -175,6 +177,11 @@ internal fun runLlvmOptimizationPipeline(context: Context) {
             // Since we are in a "closed world" internalization can be safely used
             // to reduce size of a bitcode with global dce.
             LLVMAddInternalizePass(modulePasses, 0)
+        } else if (context.config.produce == CompilerOutputKind.STATIC_CACHE) {
+            // Hidden visibility makes symbols internal when linking the binary.
+            // When producing dynamic library, this enables DCE similar to internalize but later:
```

Comment:
Maybe "..., this enables stripping unused symbols from binary with `-dead_strip` flag"?

Reply:
Makes sense.